streaming_jupyter_integrations/sql_utils.py

```diff
@@ -1,25 +1,70 @@
+from typing import Dict, Iterable, List
+
+import sqlparse
+
+DDL_KEYWORDS = {
+    'CREATE',
+    'DROP',
+    'ALTER',
+    'USE',
+    'LOAD',
+    'UNLOAD'
+}
+
+QUERY_KEYWORDS = {
+    'VALUES',
+    'WITH',  # WITH ... query
+    'SELECT'
+}
+
+DQL_KEYWORDS = {
+    *QUERY_KEYWORDS,
+    'DESCRIBE',
+    'DESC',
+    'EXPLAIN',
+    'SHOW'
+}
+
+DML_PREFIXES: Dict[str, List[str]] = {
+    'INSERT': [],
+    'EXECUTE': ['INSERT']
```

Comment:
I'm still thinking if this part is not over-engineered.
Is there any use of `EXECUTE` in Flink other than `EXECUTE INSERT`?

Reply:
Afaik no, there is no such place. It's more about tokenization — I imagine someone would write `EXECUTE   INSERT` or `EXECUTE\n  INSERT` or something like that, and so we cannot rely on simple `sql.starts_with("EXECUTE INSERT")`. So we can either write a whitespace-agnostic regex to match against, or use `sqlparse` and its tokenizator to check for the proper start. Right now it relies on the latter.
As I started to think about that, maybe I can change `DML_PREFIXES` from `Dict[str, List[str]]` to just a `List[str]` — it would be more "generic" to try to recursively walk and try to match every subsequent token 🙂

Reply:
Finally changed it to a plain regex.